nomad/event_endpoint.go

```diff
@@ -0,0 +1,210 @@
+package nomad
+
+import (
+	"context"
+	"io"
+	"time"
+
+	"github.com/hashicorp/go-msgpack/codec"
+	"github.com/hashicorp/nomad/helper"
+	"github.com/hashicorp/nomad/helper/uuid"
+	"github.com/hashicorp/nomad/nomad/stream"
+	"github.com/hashicorp/nomad/nomad/structs"
+)
+
+type Event struct {
+	srv *Server
+}
+
+func (e *Event) register() {
+	e.srv.streamingRpcs.Register("Event.Stream", e.stream)
+}
+
+func (e *Event) stream(conn io.ReadWriteCloser) {
+	defer conn.Close()
+
+	var args structs.EventStreamRequest
+	decoder := codec.NewDecoder(conn, structs.MsgpackHandle)
+	encoder := codec.NewEncoder(conn, structs.MsgpackHandle)
+
+	if err := decoder.Decode(&args); err != nil {
+		handleJsonResultError(err, helper.Int64ToPtr(500), encoder)
+		return
+	}
+
+	// forward to appropriate region
+	if args.Region != e.srv.config.Region {
+		err := e.forwardStreamingRPC(args.Region, "Event.Stream", args, conn)
+		if err != nil {
+			handleJsonResultError(err, helper.Int64ToPtr(500), encoder)
+		}
+		return
+	}
+
+	// ACL check
+	// TODO(drew) ACL checks need to be per topic
+	// All Events        Management
+	// System Events     Management
+	// Node Events       NamespaceCapabilityReadEvents
+	// Job/Alloc Events  NamespaceCapabilityReadEvents
+	if aclObj, err := e.srv.ResolveToken(args.AuthToken); err != nil {
+		handleJsonResultError(err, nil, encoder)
+		return
+	} else if aclObj != nil && !aclObj.IsManagement() {
```

Comment:
If you don't want to refine this before merging to master just make sure to file an issue to track the future work.